addon-test-support/build-owner.js

```diff
@@ -0,0 +1,104 @@
+/* globals requirejs */
+import global from './global';
+
+import ApplicationInstance from '@ember/application/instance';
+import Application from '@ember/application';
+import EmberObject from '@ember/object';
+
+import require from 'require';
+import Ember from 'ember';
+import { getResolver } from './resolver';
+
+const Owner = EmberObject.extend(
+  Ember._RegistryProxyMixin,
+  Ember._ContainerProxyMixin
+);
+
+// different than the legacy-0-6-x version (build-registry.js)
+// in the following ways:
+//
+// * fewer fallbacks (supports only Ember 2.4+)
+// * returns "owner" (instead of a POJO with registry/container)
+// * directly calls getResolver if one was not provided
+export default function(resolver = getResolver()) {
+  let fallbackRegistry, registry, container;
+  let namespace = EmberObject.create({
+    Resolver: {
+      create() {
+        return resolver;
+      },
+    },
+  });
+
+  function register(name, factory) {
+    let thingToRegisterWith = registry || container;
+
+    if (
+      !(container.factoryFor
+        ? container.factoryFor(name)
+        : container.lookupFactory(name))
+    ) {
+      thingToRegisterWith.register(name, factory);
+    }
+  }
+
+  fallbackRegistry = Application.buildRegistry(namespace);
+  fallbackRegistry.register('component-lookup:main', Ember.ComponentLookup);
+
+  registry = new Ember.Registry({
+    fallback: fallbackRegistry,
+  });
+
+  // Ember.ApplicationInstance was exposed in Ember 2.8
+  if (ApplicationInstance && ApplicationInstance.setupRegistry) {
+    ApplicationInstance.setupRegistry(registry);
+  }
+
+  // these properties are set on the fallback registry by `buildRegistry`
+  // and on the primary registry within the ApplicationInstance constructor
+  // but we need to manually recreate them since ApplicationInstance's are not
+  // exposed externally
+  registry.normalizeFullName = fallbackRegistry.normalizeFullName;
+  registry.makeToString = fallbackRegistry.makeToString;
+  registry.describe = fallbackRegistry.describe;
+
+  let owner = Owner.create({
+    __registry__: registry,
+    __container__: null,
+  });
+
+  container = registry.container({ owner: owner });
+  owner.__container__ = container;
+
+  // TODO: this manual Ember Data setup should be removed in favor of
+  // running the applications `initializers` to ensure the container is
+  // properly setup, this would only need to be done once per test suite
```

Comment:
running the initializers automatically would change the semantic of the tests though which might make the migration path a little harder 🤔

Reply:
Ya, my thinking is basically that we should _not_ need to hard code customized setup for ember-data here.  There are basically two options to accomplish that goal:

1. Run initializers (but not instance-initializers) _only once_ for the full suite, just to build up the fallback registry for lookups (this essentially emulates the way fastboot works for example by creating a single registry that is shared amongst many different app instances).
2. Remove the need for ember-data to have these initializers.  

I think both are actually valid options...

Reply:
second option seems easier to me

maybe we could also make it configurable via an option if the initializers should be run for the test or not?

Reply:
@Turbo87 - Just to clarify (because your last comment seemed to indicate that we aren't on the same page), I am **not** suggesting that we run initializers for each test.  I am saying that we should run them once for the entire test suite, then use the resulting registry as our fallback registry in this script (notice how we setup the fallback registry system here?).  The only purpose for `initializers` at this point (since sometime late in 1.x) is to setup the registry (e.g. add automatic injections, register manual factories, etc), this "registry setup" work is _exactly_ what we actually want to run.  Conversely, the `instance-initializers` are specifically intended to be able to work with instances (e.g. they can `.lookup`) from the container, which is _exactly_ what we want to prevent.

Reply:
> The only purpose for initializers at this point (since sometime late in 1.x) is to setup the registry

Yeah, guess I'm just a little worried about people misusing these things for other purposes